### PR TITLE
Add feature tests for "Additional Code" and "Footnote"

### PR DESCRIPTION
**Pull request by steveholmes**

### CODE REVIEW

File: spec/features/workbasket/footnotes/footnotes_creation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    _user = create(:user)
+
+    visit(root_path)
+    click_on("Create a new footnote")
+
+    within(find("fieldset", text: "What type of footnote is this?")) do
+      select_dropdown_value(footnote_type.footnote_type_id)
+    end
+
+    # save_and_open_page
```
Comment (KyleMacPherson) [inline spec/features/workbasket/footnotes/footnotes_creation_spec.rb]: [minor] left a pesky`save_and_open_page` in

File: spec/features/workbasket/footnotes/footnotes_creation_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+require "rails_helper"
+
+RSpec.describe "adding footnotes", :js do
+
+  let!(:footnote_type) { create(:footnote_type) }
+  let!(:footnote_type_description) { create(:footnote_type_description, footnote_type_id: footnote_type.footnote_type_id) }
```
Comment (KyleMacPherson) [inline spec/features/workbasket/footnotes/footnotes_creation_spec.rb]: Do we need this to select from the dropdown? It looks like we select by footnote type id at first glance

File: spec/features/workbasket/footnotes/footnotes_creation_spec.rb (hunk 1 above)
Comment (steveholmes) [reply spec/features/workbasket/footnotes/footnotes_creation_spec.rb]: Oops

File: spec/features/workbasket/footnotes/footnotes_creation_spec.rb (hunk 2 above)
Comment (steveholmes) [reply spec/features/workbasket/footnotes/footnotes_creation_spec.rb]: yes, type description is also needed for the dropdown selector.

**Pull request merged**